wordpress-mobile/WordPress-iOS
# Update synced media references when saving a post with ongoing uploads

Fixes https://github.com/WordPress/gutenberg/issues/33593

**To test:**
1. Create a post.
1. Add a gallery block and tap "Add Media".
1. Select "Choose from device" option.
1. Select multiple images from the device and confirm the selection.
1. While images are uploading, leave the editor.
1. Wait for uploads to complete while in the post list.
1. Re-open the post with the gallery block.
1. Tap on the post settings button (three dots button located at the top).
1. Tap on "Switch to HTML mode"
1. Observe that the `src` attributes of all `img` tags reference a remote URL and not a local version.
    - Remote URLs start with `https://`
    - Local URLs start with `///`

## Regression Notes
1. Potential unintended areas of impact
The fix is specific to the post save process when there are uploads in progress, so apart from this area, I don't expect other unintended areas of impact.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested in the following scenarios for this area:
- Save a post with completed and pending media uploads
- Save a post with only pending media uploads
- Save a post with no media uploads

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
